### ISSUE TITLE
Bug Report: Chart Editor Has A Maximum Song Length of 25 Minutes and 21.742 Seconds

### ISSUE THREAD

**Issue by Shaunyman** [FunkinCrew/Funkin#3070]
### Issue Checklist

- [x] I have properly named the issue
- [X] I have checked the issues/discussions pages to see if the issue has been previously reported

### What platform are you using?

Itch.io (Downloadable Build) - Windows

### If you are playing on a browser, which one are you using?

None

### Version

0.4.1

### Context (Provide images, videos, etc.)

The chart editor will not allow a song longer than 25 minutes and 21.742 seconds. Even one millisecond longer and the chart editor becomes very small (less than one measure) and you cannot add notes for the whole song. Attached are two photos, one of them is a song that is 25 minutes and 21.742 seconds. It looks like a normal chart would. The other is of a song that is 25 minutes and 21.743 seconds (only one millisecond longer). It does not look like a normal chart would. Also, the times on the bottom right and left of the chart are both negative and only a couple of minutes instead of the whole length. 
![25m 21 742s](https://github.com/user-attachments/assets/d549b5d1-f737-4925-aa93-f610a4076fa5)
![25m 21 743s](https://github.com/user-attachments/assets/2525c811-e7a5-4f1f-848c-77f4a8105e28)

### Steps to reproduce (or crash logs, errors, etc.)

Go into chart editor. Select File -> New Chart. In "Create New" select any of the 3 options (the issue occurs regardless of which difficulty settings you choose). Click "Continue". Select a song that is longer than 25 minutes and 21.742 seconds and the chart editor will break. Select a song that is 25 minutes and 21.742 seconds or less and the chart editor will work as intended.

**Comment by Hundrec** [comment]
Very interesting bug and precise boundary!

Does changing the BPM (or any other song properties) affect this boundary?

**Comment by Shaunyman** [comment]
After applying the following changes, nothing changed with the boundary: setting the BPM to 50, 200, and 150, testing all of the available stages, and changing the note style to pixel notes. Something to be noted is that I did not attempt changing the 3 character options to see if those would cause a change in the boundary. So, while unlikely, it is possible that changing any of those things could cause the boundary to increase, resulting in the ability to make a chart longer than 25 minutes and 21.742 seconds.

**Comment by NotHyper-474** [comment]
I tried to make a song with that same length and it glitched either way. Tried exporting with a different sample rate and it worked.
I'm thinking it's a integer limit issue somewhere.

**Comment by NotHyper-474** [comment]
After a bit of debugging I found out that the issue traces all the way around to these lines in OpenFL's sound backend

https://github.com/FunkinCrew/openfl/blob/228c1b5063911e2ad75cef6e3168ef0a4b9f9134/src/openfl/media/Sound.hx#L675-L679

And as I imagined it has to do with integer limitations. Just look at how the math goes

```
samples = (268,435,084 * 8) / (2 * 16)
samples = 2,147,480,672 / 32
samples = 67,108,771
lengthMs = 67,108,771 / sampleRate (44,100) * 1000 ≈ 1,521,740.84 ms
```
2.147.480.672 is pretty close (2475 of difference) to the 32-bit integer limit (2.147.483.647) so that's why it works.

Now I debugged the same song with a sample rate of 48,000 Hz
`samples = (292,174,384 * 8) / 32`, this is where we have a problem. 292,174,384 times 8 is 2,337,395,072, way bigger than that limit. So the integer overflows towards negative and fucks with everything.
In the end we get this:
```
samples = (292,174,384 * 8) / 32
samples = -1,957,572,224 / 32
samples = -61,174,132
lengthMs = -61,174,132 / 48,000 * 1000
lengthMs ≈ -1,274,461.08 ms
```

**Comment by Hundrec** [comment]
Amazing detective work, Hyper!

That pesky integer limit messes with everything!

**Comment by NotHyper-474** [comment]
Issue seems to have been fixed on 0.5.0

**Comment by Hundrec** [comment]
If this is fixed as of 0.5.2, please close this issue.
We've got bigger (longer) problems to tackle...

**Comment by Shaunyman** [comment]
> If this is fixed as of 0.5.2, please close this issue. We've got bigger (longer) problems to tackle...

Didn't even realize that was something I could do! Thought that was something done by the dev team.
